extensions/markdown-language-features/src/languageFeatures/copyFiles/copyPasteLinks.ts

```diff
@@ -45,12 +46,12 @@ class PasteLinkEditProvider implements vscode.DocumentPasteEditProvider {
 	}
 }
 
-export function validateLink(urlList: string): boolean {
-	const url = urlList?.split(/\s+/);
-	if (url.length > 1 || !externalUriSchemes.includes(vscode.Uri.parse(url[0]).scheme)) {
-		return false;
-	}
-	return true;
+// filter out white spaces
+export function validateLink(urlList: string): { isValid: boolean; cleanedUrlList: string } {
+	const url = urlList?.split(' ').filter(item => item !== '');
```

Comment:
Try using `.trim()` instead. The cases we want to support are when there are trailing or leading whitespaces, and trim should handle that nicely

Reply:
The new validateLink method is updated. 

uri.parse throws an error if there is a space before the link (text before the link) but not after. So the new validateLink method considers this case. But if there are words after the link, uri.parse will throw. That's why I'm splitting the string of urlList to see if there is other content (like text), making the "link" invalid.